plain/src/main/java/com/esewa/internship/gyalbusherpa/ProductUnitTestingInJUnit4/src/main/java/org/dao/ProductDao.java

```diff
@@ -0,0 +1,62 @@
+package org.dao;
+
+import org.entity.Product;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ProductDao {
+
+    private final List<Product> productList;
+
+    public ProductDao() {
+        productList = new ArrayList<>();
+    }
+
+    public Product getProductById(int id) {
+        for (Product product : productList) {
+            if (product.getId() == id) {
+                return product;
+            }
+        }
+        return null;
```

Comment:
Never return null. Instead, return Optional.empty(). That way you can be sure that the person using this method handles NPE.

Reply:
Optional used instead of returning null